JCSDA/spack-stack
# Update Parallel Works configs (AWS, Azure, Gcloud)

### Summary

This PR updates the Parallel Works spack-stack site configurations so that they are consistent with each other, with the directory layout described in the project charter, and provide everything needed for UFS and JEDI-Skylab applications.

Note that in order to function correctly, the following script needs to be run every time when a new cluster (head node) is provisioned (this can be run manually or automatically, configured on the Parallel Works resources page):
```
> cat /contrib/admin/basic_setup.sh
#!/bin/bash

chmod 777 /contrib
yum install -y qt5-qtbase-devel
yum install -y qt5-qtsvg-devel
yum install -y xorg-x11-xauth
yum install -y xorg-x11-apps
yum install -y perl-IPC-Cmd
yum install -y gettext-devel
```

**To be discussed**: Is this updated site config acceptable for EPIC, or do we need to different site configurations for EPIC and JCSDA? If we use one shared config, how do we make sure that we test updates and keep the configs in sync, given that we cannot mount the same filesystem under `/contrib`? Rsync the entire `/contrib/spack-stack` folder to somewhere else? Or build on a completely separate drive in the future that we can mount in both the EPIC and JCSDA accounts, for example as `/contrib-shared`?

**Note 1**: I updated the documentation (pre-configured sites, maintainers section) with my changes. I decided to put the basic setup instructions in a `README.md` (similar to what I did for `aws-pcluster` in the past), because there are a few more steps needed than on the HPCs. I think this works better for the cloud platforms.

**Note 2**: With this setup, I was able to build `wgrib2` on all platforms (addressing https://github.com/JCSDA/spack-stack/issues/569).

### Testing

- Install spack-stack develop as of August 17
    - [x] NOAA-AWS
    - [x] NOAA-Azure
    - [x] NOAA-Gcloud
- Build jedi-bundle develop as of August 21, run ctests. make sure all tests pass
    - [x] NOAA-AWS
    - [x] NOAA-Azure
    - [x] NOAA-Gcloud
- Build ufs-weather-model / ufs-srw-eapp (@ulmononian) and run basic tests (no failures)
    - [x] NOAA-AWS
    - [x] NOAA-Azure
    - [x] NOAA-Gcloud
- Build remaining Skylab components as of August 21 and run the following Skylab experiments: `qg-fullDA.yaml`, `l95-fullDA.yaml`, `skylab-atm-land-small.yaml`
    - [x] NOAA-AWS
    - [x] NOAA-Azure
    - [x] NOAA-Gcloud

See also https://github.com/JCSDA-internal/jedi-bundle/issues/31 if you have access.

### Applications affected

n/a

### Systems affected

NOAA Parallel Works platforms (AWS, Azure, Gcloud)

### Dependencies

n/a

### Issue(s) addressed

Resolves https://github.com/JCSDA-internal/jedi-bundle/issues/31
Resolves https://github.com/JCSDA/spack-stack/issues/569

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~
